src/main/java/aurora/carevisionapiserver/domain/camera/repository/CustomCameraRepositoryImpl.java

```diff
@@ -17,19 +24,41 @@ public class CustomCameraRepositoryImpl implements CustomCameraRepository {
     private final JPAQueryFactory queryFactory;
 
     @Override
-    public List<Camera> findAllCamerasSortedByBed(long departmentId) {
-        QBed bed = QBed.bed;
+    public Slice<Camera> findAllCamerasSortedByBed(Department department, Long lastIdx, int size) {
         QCamera camera = QCamera.camera;
-        return queryFactory
-                .selectFrom(camera)
-                .leftJoin(bed)
-                .on(camera.bed.id.eq(bed.id))
-                .where(camera.bed.department.id.eq(departmentId))
-                .orderBy(
-                        bed.inpatientWardNumber.asc(),
-                        bed.patientRoomNumber.asc(),
-                        bed.bedNumber.asc())
-                .fetch();
+        QBed bed = QBed.bed;
+
+        List<Camera> cameras =
+                queryFactory
+                        .select(camera)
+                        .from(camera)
+                        .leftJoin(bed)
+                        .on(bed.camera.eq(camera))
+                        .leftJoin(QDepartment.department)
+                        .on(bed.department.eq(department))
+                        .where(
+                                Expressions.numberTemplate(
+                                                Long.class,
+                                                "regexp_replace({0}, '^[^0-9]+', '')",
+                                                camera.id)
+                                        .gt(lastIdx))
```

Comment:
정규식으로 숫자로 된 ID 뽑아내는 부분 메서드로 추출하면 더 깔꼼할 것 같습니다!! 요런 식으루 제안해봄니다
```suggestion
                                .where(
                                extractNumericPart(camera)
                                        .gt(lastIdx))
                                
                                
                                
private static NumberTemplate<Long> extractNumericPart(QCamera camera) {
        return Expressions.numberTemplate(
                Long.class,
                "regexp_replace({0}, '^[^0-9]+', '')",
                camera.id);
    }
```


```java
    private static NumberTemplate<Long> extractNumericPart(QCamera camera) {
        return Expressions.numberTemplate(
                Long.class,
                "regexp_replace({0}, '^[^0-9]+', '')",
                camera.id);
    }
```

Reply:
좀 더 가독성 좋은거같아요! 수정하겠습니다
아래 두 코멘트도 아주 굿이어서 수정해서 올릴게용!